### ISSUE TITLE
Add @testing-library/react and jest-dom

### ISSUE THREAD

**Issue by thektan** [liferay/liferay-npm-tools#108]
Proposal to add the following devDependencies:
- @testing-library/react (https://github.com/testing-library/react-testing-library)
- jest-dom (https://github.com/testing-library/jest-dom)

Both of these libraries provide useful utilities while writing tests.

**React testing library** allows for easier testing in the DOM and I currently agree with the philosophy of the library: "The more your tests resemble the way your software is used, the more confidence they can give you." Using the library you'd test for more specific values like if a button is displaying the correct text rather than relying on a snapshot of the markup which can easily mask erroneous tests that you'd constantly have to update. 

**jest-dom** provides additional custom matchers like `toBeInTheDocument` which checks if an element is in the document  and `toHaveTextContent` which has been handy in checking if an element has certain text (rather than making sure the text is wrapped in a `<div>` or has a certain css class, etc).

In summary: Compared to [jest snapshot testing](https://jestjs.io/docs/en/snapshot-testing), both provide better maintainability and bug discoverability in tests.

This is already being used in `segments-web` (https://github.com/liferay/liferay-portal/blob/master/modules/apps/segments/segments-web/package.json#L26) and I've been using it in a development branch for `portal-search-ranking-web` (https://github.com/arboliveira/liferay-portal/blob/LPS-89524_Search_Tuning_Result_Rankings/modules/apps/portal-search/portal-search-ranking-web/package.json#L64)

cc: @oliv-yu @andresfulla

**Comment by andresfulla** [comment]
Both tools have been very useful for us in `segments-web`. Writing tests has been relatively easy and we have more confidence applying any change. Both libraries are hight quality and have healthy communities.

I don't think we are making the most of it, so the more people uses it within the company, the more value this gives us.

@dgarciasarai has been using it too

**Comment by dgarciasarai** [comment]
I think that's a great idea! :)
I have been using it for a few months since I started working with React and I was capable of writing tests easily. We can't forget to check the tests before sending new features or bug fixes because, with that little action, we can avoid future bugs and unexpected behavior.

Thanks @thektan! 👍

**Comment by wincent** [comment]
@andresfulla or @dgarciasarai: can one of you link to what you think is an example of a good test that you've written?

I personally haven't used ~`react-testing-library`~ `@testing-library/react` but from what I've seen of it it looks pretty good.

**Comment by andresfulla** [comment]
I wouldn't tag any of them as good, but:
- [SegmentsEdit.es.js](https://github.com/andresfulla/liferay-portal/blob/master/modules/apps/segments/segments-web/test/js/components/segment_edit/SegmentEdit.es.js) helps us control initial state changes.
- [LocalizedInput.es.js](https://github.com/andresfulla/liferay-portal/blob/master/modules/apps/segments/segments-web/test/js/components/title_editor/LocalizedInput.es.js) covers some interactions with the component.

**Comment by wincent** [comment]
@thektan: I turned the title prefix into an actual tag and I'll update the guidelines accordingly.

**Comment by thektan** [comment]
@jbalsas @wincent What would be the next steps to be able to reach some sort of conclusion? I can send a PR to add these to portal whenever I get the go-ahead (or start refactoring if we decide not to add these).

**Comment by jbalsas** [comment]
Hey @thektan, thanks for the reminder! I think this got lost here as we're trying to move our standardization process to https://github.com/liferay/liferay-frontend-guidelines/tree/master/guidelines.

I've filed https://github.com/liferay/liferay-frontend-guidelines/issues/49 so we can work on that. We will be working on the next couple of weeks on the react developer experience in DXP, so I think we should be able to provide both an answer and a solution in the next couple of weeks.

I linked the guidelines issue to this conversation, so these should be taken into account and we can move over there if there's anything else to add.

Thanks a lot for your patience and your input @thektan!
